src/esi/Expression.cc

```diff
@@ -97,6 +100,16 @@ stackpop(stackmember * s, int *depth)
     cleanmember(&s[*depth]);
 }
 
+/// push item onto the stack.
+/// Throws an exception with the given message if depth is not referencing
+/// a valid location of the stack both before and after the push.
+static void
+stackpush(stackmember *stack, stackmember &item, int *depth, const char *message)
```

Comment:
Overall, it is usually a bad idea to pass context specifics to inner methods just so that they can throw a more detailed error message. Context reporting or annotation, if it is really needed, should be done by the caller that knows what operation it is processing. I doubt we need to be very precise here, but if we do, we can probably store the last operation name in some place where the exception handler can find it.

I do not insist on removing (context from) the `message` parameter but please consider it.

Reply:
You _required_ that these *two* lines be extracted to de-duplicate - one of those lines was the caller producing this caller-specific thrown message. Without this parameter the de-duplication was a pointless waste of time and effort.

Reply:
To save time, I am going to ignore your comment and resolve this change request because this change request is not blocking, and your comment appears to misinterpret some requirements to arrive at an absurd conclusion.